smart-contract/assembly/contracts/internals/file-list.ts

```diff
@@ -0,0 +1,34 @@
+import { FILES_PATH_LIST } from '../utils/const';
+
+export function _pushFilePath(filePath: string): void {
+  const filePathList = FILES_PATH_LIST.mustValue();
+  filePathList.push(filePath);
+  FILES_PATH_LIST.set(filePathList);
+  _getFilePathList();
+}
+
+export function _removeFilePath(filePath: string): void {
+  const filePathList = FILES_PATH_LIST.mustValue();
+  if (!_isPathFileInList(filePath)) return;
+  // TODO: check if it works
+  FILES_PATH_LIST.set(filePathList.filter((path) => path !== filePath));
+}
+
+export function _getFilePathList(): string[] {
+  const result = FILES_PATH_LIST.tryValue();
+  if (result.error) {
+    FILES_PATH_LIST.set([]);
+    return [];
+  }
+  return result.unwrap();
+}
+
+// HELPERS
+// TODO - Can cost Gas, how to improve it?
```

Comment:
maybe overkill but a hashtable could be good here, with the keys we could do a binary search to be even more efficient

Reply:
@thomas-senechal ?

Reply:
this could also solve the randomlyGenereated name problem

Reply:
We have that: https://github.com/massalabs/massa-as-sdk/blob/main/assembly/helpers/mapKeyManager.ts

But I'm unsure it's cost effective as we won't store anything in this map ? Just use the keys ? 
I guess the best would be to check the underlying code of the map to see if it's cost effective to use that or the list

Reply:
Note that the idea was also to have something simple enough so we can simply read the datastore from the server instead of doing a readonly call

Reply:
Maybe we could do that in a v2 or in an improvement phase